app/scripts/modules/amazon/src/function/details/AmazonFunctionDetails.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+import {
+  Details,
+  CollapsibleSection,
+  IFunction,
+  Application,
+  AccountTag,
+  IOverridableProps,
+  Overrides,
+} from '@spinnaker/core';
+import { IAmazonFunctionSourceData, IAmazonFunction } from 'amazon/domain';
+import { FunctionActions } from './FunctionActions';
+import { AwsReactInjector } from 'amazon/reactShims';
+import * as _ from 'lodash';
+
+export interface IFunctionFromStateParams {
+  account: string;
+  region: string;
+  functionName: string;
+}
+
+export interface IAmazonFunctionDetailsProps extends IOverridableProps {
+  app: Application;
+  functionObj: IFunction;
+}
+
+@Overrides('function.details', 'aws')
+export class AmazonFunctionDetails extends React.Component<IAmazonFunctionDetailsProps, any> {
+  constructor(props: IAmazonFunctionDetailsProps) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  public extractFunction(): void {
+    const { app, functionObj: functionFromProps } = this.props;
+    const functionDef: IFunction = app.functions.data.find((test: IFunction) => {
+      return (
+        test.functionName === functionFromProps.functionName &&
+        test.region === functionFromProps.region &&
+        test.account === functionFromProps.account
+      );
+    });
+
+    if (functionDef) {
+      AwsReactInjector.functionReader
```

Comment:
race condition here, let's use the `this.destroy$` observable pattern since this is a class component.